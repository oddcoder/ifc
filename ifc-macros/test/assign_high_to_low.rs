use ifc_macros::ifc_block;

fn main() {
    ifc_block!{
        let x = 5;
        #[IFC(Low)]
        let _y;
        _y = x;
    };
}