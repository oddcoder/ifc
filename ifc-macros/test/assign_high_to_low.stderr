error: Cannot assign high expression to low variable
 --> $DIR/assign_high_to_low.rs:8:9
  |
8 |         _y = x;
  |         ^^^^^^
  |
help: Low Variable
 --> $DIR/assign_high_to_low.rs:8:9
  |
8 |         _y = x;
  |         ^^
help: High Expression
 --> $DIR/assign_high_to_low.rs:8:14
  |
8 |         _y = x;
  |              ^
